File: apps/web/src/features/incomes/api/list-incomes.ts
import { api } from '@/api/client'
import type { IncomePage } from '@/features/incomes/income'

export async function listIncomes(month: string, limit: number, offset: number): Promise<IncomePage> {
  const { data, error } = await api.GET('/api/v1/incomes', {
    params: { query: { month, limit, offset } },
  })
  if (error) throw new Error('Failed to load incomes')
  return data
}

File: apps/api/internal/incomes/application/list_incomes.go
package application

import (
	"context"

	"caudal-api/internal/incomes/ports"
	shared "caudal-api/internal/shared/domain"
)

type ListIncomes struct {
	repository ports.Repository
}

func NewListIncomes(repository ports.Repository) ListIncomes {
	return ListIncomes{repository: repository}
}

func (useCase ListIncomes) Execute(ctx context.Context, query ListIncomesQuery) (ports.Page, error) {
	start, end := shared.FinancialMonthBounds(query.Year, query.Month)
	return useCase.repository.List(ctx, start, end, query.Limit, query.Offset)
}
